### ISSUE TITLE
CMAKE MAD_EXCEPTION = assert causes compilation error

### ISSUE THREAD

**Issue by pchong90** [m-a-d-n-e-s-s/madness#183]
example error message
```
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/world.cc:38:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/world.h:68:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/worldmpi.h:55:
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/safempi.h:621:13: error:
      use of undeclared identifier '__assert_rtn'; did you mean 'madness::__assert_rtn'?
            MADNESS_ASSERT(pimpl);
            ^
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/madness_exception.h:144:37: note:
      expanded from macro 'MADNESS_ASSERT'
#  define MADNESS_ASSERT(condition) assert(condition)
                                    ^
/usr/include/assert.h:93:34: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) ...
                                 ^
/usr/include/assert.h:76:6: note: 'madness::__assert_rtn' declared here
void __assert_rtn(const char *, const char *, int, const char *) __dead2;
     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

**Comment by pchong90** [comment]
@evaleev I still got `use of undeclared identifier 'assert'` when I built it with TiledArray

I did use `-DMADNESS_TAG` to point to the newest madness commit.

example of error message
```
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/future.cc:38:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/future.h:45:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/dependency_interface.h:44:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/world.h:68:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/worldmpi.h:55:
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/safempi.h:615:13: error:
      use of undeclared identifier 'assert'
            MADNESS_ASSERT(pimpl);
            ^
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/madness_exception.h:146:37: note:
      expanded from macro 'MADNESS_ASSERT'
#  define MADNESS_ASSERT(condition) assert(condition)
                                    ^
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/future.cc:38:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/future.h:45:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/dependency_interface.h:44:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/world.h:68:
In file included from /Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/worldmpi.h:55:
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/safempi.h:621:13: error:
      use of undeclared identifier 'assert'
            MADNESS_ASSERT(pimpl);
            ^
/Users/ChongPeng/Workspace/Development/source/tiledarray/external/src/madness/src/madness/world/madness_exception.h:146:37: note:
      expanded from macro 'MADNESS_ASSERT'
#  define MADNESS_ASSERT(condition) assert(condition)
                                    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```

**Comment by evaleev** [comment]
resolved by 20f2f61e65af3cb0b0911441f4fee7b62cb9cc97

**Comment by pchong90** [comment]
Thank you. Compiled successfully.
